satpy/readers/fci_l2_nc.py

```diff
@@ -401,3 +401,58 @@ def _modify_area_extent(stand_area_extent):
         area_extent = tuple([ll_x, ll_y, ur_x, ur_y])
 
         return area_extent
+
+class FciL2NCAMVFileHandler(FciL2CommonFunctions, BaseFileHandler):
+    """Reader class for FCI L2 AMV products in NetCDF4 format."""
+    def __init__(self, filename, filename_info, filetype_info):
+        """Open the NetCDF file with xarray and prepare for dataset reading."""
+        super().__init__(filename, filename_info, filetype_info)
+
+        # Use xarray's default netcdf4 engine to open the file
+        self.nc = xr.open_dataset(
+            self.filename,
+            decode_cf=True,
+            mask_and_scale=True,
+            chunks={
+                "number_of_images": CHUNK_SIZE,
+                # 'number_of_height_estimates': CHUNK_SIZE,
+                "number_of_winds": CHUNK_SIZE
+            }
+        )
```

Comment:
This could be made into a cached property for lazy reading (ie the file will not be open until actually needed).

Reply:
I am only mimicking the rest of the File Handles here (I could remove the commented part). 
```python 
class FciL2NCFileHandler(FciL2CommonFunctions, BaseFileHandler):
    """Reader class for FCI L2 products in NetCDF4 format."""

    def __init__(self, filename, filename_info, filetype_info, with_area_definition=True):
        """Open the NetCDF file with xarray and prepare for dataset reading."""
        super().__init__(filename, filename_info, filetype_info)

        # Use xarray's default netcdf4 engine to open the fileq
        self.nc = xr.open_dataset(
            self.filename,
            decode_cf=True,
            mask_and_scale=True,
            chunks={
                "number_of_columns": CHUNK_SIZE,
                "number_of_rows": CHUNK_SIZE
            }
        )
```

Reply:
Yes, I know that's the way it was done elsewhere, but it could be an improvement :)

Reply:
well, TBH I don't know how to do that. 
And given the size of AMVs file I would not see the real benefit of it. 
We can chat about that later on Slack if you want to show me how to do it.

Reply:
```suggestion

@cached_property
def nc(self)
        # Use xarray's default netcdf4 engine to open the file
        self.nc = xr.open_dataset(
            self.filename,
            decode_cf=True,
            mask_and_scale=True,
            chunks={
                "number_of_images": CHUNK_SIZE,
                # 'number_of_height_estimates': CHUNK_SIZE,
                "number_of_winds": CHUNK_SIZE
            }
        )
```

Reply:
The benefit is to not to have to open the file when the FileHandler is created. I don't know exactly how this FileHandler is used, but in some cases a FileHandler is created just to be discarded without being used.